modules/queue/bytefifo.go

```diff
@@ -4,14 +4,16 @@
 
 package queue
 
+import "context"
+
 // ByteFIFO defines a FIFO that takes a byte array
 type ByteFIFO interface {
 	// Len returns the length of the fifo
-	Len() int64
+	Len(ctx context.Context) int64
```

Comment:
Since these operations are very quick, I don't think a `context.Context` is necessary, Only those functions which will has a loop or will spend many time should use a `context.Context`. And also, the interface is ugly.

Reply:
Redis V8 has ctx for this.

Reply:
https://github.com/go-redis/redis/blob/f7ac48d4d5490cbf0d166b9f42dba9e62483b77e/commands.go#L180

Reply:
yes can remember this ... :X

Reply:
This explicit passing in of the correct context also fixes an issue with Tasks left in the queue always being reported as -1